File: test_api.ts
import { GoogleGenAI } from '@google/genai';
import type { Interactions } from '@google/genai';

const ai = new GoogleGenAI({
    apiKey: process.env['GEMINI_API_KEY'],
});

const tools: Interactions.Tool[] = [
    {
        type: 'google_search',
    },
];

const generationConfig = {
    temperature: 1,
    max_output_tokens: 65536,
    top_p: 0.95,
};

async function main() {
    const interaction = await ai.interactions.create({
        model: 'models/gemini-2.5-flash',
        input: 'Hello',
        tools: tools,
        generation_config: generationConfig,
    });

    console.log(interaction.steps?.at(-1));
}

main();

File: test_api.py
import os
from google import genai

client = genai.Client(
    api_key="AQ.Ab8RN6K1qMEKOHqNQUtmJ38F1BoK-QPBgenOEugDe18BBtekWQ"
)

tools = [
    {
        'type': 'google_search',
    },
]

generation_config = {
    'temperature': 1,
    'max_output_tokens': 65536,
    'top_p': 0.95,
}

interaction = client.interactions.create(
    model='models/gemini-3.1-flash-lite',
    input='',
    tools=tools,
    generation_config=generation_config,
)

print(interaction.steps[-1])


